lib/filterx/func-unset-empties.c

```diff
@@ -0,0 +1,316 @@
+  filterx_object_unref(obj);
+  return NULL;
+}
+
+static void
+_free(FilterXExpr *s)
+{
+  FilterXFunctionUnsetEmpties *self = (FilterXFunctionUnsetEmpties *) s;
+  filterx_expr_unref(self->object_expr);
+  filterx_function_free_method(&self->super);
+}
+
+static FilterXExpr *
+_extract_object_expr(FilterXFunctionArgs *args, GError **error)
+{
+  FilterXExpr *object_expr = filterx_function_args_get_expr(args, 0);
+  if (!object_expr)
+    {
+      g_set_error(error, FILTERX_FUNCTION_ERROR, FILTERX_FUNCTION_ERROR_CTOR_FAIL,
+                  "argument must be set: object. " FILTERX_FUNC_UNSET_EMPTIES_USAGE);
+      return NULL;
+    }
+
+  return object_expr;
+}
+
+static gboolean
+_extract_optional_literal_bool_arg(FilterXFunctionArgs *args, const gchar *arg_name, gboolean *output, GError **error)
+{
+  gboolean exists, eval_error;
+  gboolean value = filterx_function_args_get_named_literal_boolean(args, arg_name, &exists, &eval_error);
+  if (!exists)
+    return TRUE;
+
+  if (eval_error)
+    {
+      g_set_error(error, FILTERX_FUNCTION_ERROR, FILTERX_FUNCTION_ERROR_CTOR_FAIL,
+                  "%s argument must be boolean literal. " FILTERX_FUNC_UNSET_EMPTIES_USAGE, arg_name);
+      return FALSE;
+    }
+
+  *output = value;
+  return TRUE;
+}
+
+static gboolean
+_extract_optional_args(FilterXFunctionUnsetEmpties *self, FilterXFunctionArgs *args, GError **error)
+{
+  return _extract_optional_literal_bool_arg(args, "recursive", &self->recursive, error) &&
+         _extract_optional_literal_bool_arg(args, "string", &self->drop_strings, error) &&
+         _extract_optional_literal_bool_arg(args, "null", &self->drop_nulls, error) &&
+         _extract_optional_literal_bool_arg(args, "number", &self->drop_numbers, error) &&
+         _extract_optional_literal_bool_arg(args, "dict", &self->drop_dicts, error) &&
+         _extract_optional_literal_bool_arg(args, "list", &self->drop_lists, error);
+}
+
+static gboolean
+_extract_args(FilterXFunctionUnsetEmpties *self, FilterXFunctionArgs *args, GError **error)
+{
+  if (filterx_function_args_len(args) != 1)
+    {
+      g_set_error(error, FILTERX_FUNCTION_ERROR, FILTERX_FUNCTION_ERROR_CTOR_FAIL,
+                  "invalid number of arguments. " FILTERX_FUNC_UNSET_EMPTIES_USAGE);
+      return FALSE;
+    }
+
+  self->object_expr = _extract_object_expr(args, error);
+  if (!self->object_expr)
+    return FALSE;
+
+  if (!_extract_optional_args(self, args, error))
+    return FALSE;
+
+  return TRUE;
+}
+
+FilterXFunction *
+filterx_function_unset_empties_new(const gchar *function_name, FilterXFunctionArgs *args, GError **error)
+{
+  FilterXFunctionUnsetEmpties *self = g_new0(FilterXFunctionUnsetEmpties, 1);
+  filterx_function_init_instance(&self->super, function_name);
+  self->super.super.eval = _eval;
+  self->super.super.free_fn = _free;
+
+  self->recursive = TRUE;
+  self->drop_strings = self->drop_nulls = self->drop_numbers = self->drop_dicts = self->drop_lists = TRUE;
```

Comment:
I don't tihnk dropping numbers should be enabled by default, 0 is not an unset value.

Also, it would make sense to add a "null" string value (e.g. "n/a" or "-" are often used to indicate unset values)